wherehows-web/app/components/search/containers/search-facets.ts

```diff
@@ -0,0 +1,162 @@
+import Component from '@ember/component';
+import { get } from '@ember/object';
+import { computed } from '@ember-decorators/object';
+import { capitalize } from '@ember/string';
+import { task } from 'ember-concurrency';
+import { DatasetPlatform } from 'wherehows-web/constants';
+import { IDataPlatform } from 'wherehows-web/typings/api/list/platforms';
+import { readPlatforms } from 'wherehows-web/utils/api/list/platforms';
+import { arrayMap } from 'wherehows-web/utils/array';
+import { IFacetsSelectionsMap, IFacetsCounts } from 'wherehows-web/utils/api/search';
+
+/**
+ * Options inside a facet
+ * @interface ISearchFacetOption
+ */
+interface ISearchFacetOption {
+  value: string;
+  label: string;
+}
+
+/**
+ * Interface of a facet
+ */
+interface ISearchFacet {
+  name: string;
+  displayName: string;
+  values: Array<ISearchFacetOption>;
+}
+
+/**
+ * Container component for search facets
+ * It will store state related to search facets.
+ */
+export default class SearchFacetsContainer extends Component {
+  didInsertElement(this: SearchFacetsContainer) {
```

Comment:
for the basic component hooks, isn't the explicit this declaration unnecessary?

Reply:
Maybe, that was there, so I haven't tried to improve that.

Reply:
You are right.